### DISCUSSION TITLE
PySide6 upgrade

### DISCUSSION THREAD

**Discussion by joshy** [abey79/vsketch#174] [General]
Hi Antoine,

as PySide2 is not supported on Apple M1 I am trying to migrate it to PySide6. Let's see how far I come and if successful I can submit a PR.

Cheers,
Joshy

**Comment by joshy** [comment]
Oh bummer, not even PySide6 can be installed for Apple M1. So this discussion can be deleted. Sorry.

**Reply by abey79** [reply]
Actually, there is a real need to migrate to PySide6. Proper M1 support will be added to it within the next couple of release, whereas Qt5/PySide2 is hardly getting any love any more. The first step would be to migrate vpype's viewer to PySide6, as vsketch heavily reuses it.

I already have a branch where most of the easy stuff is done: https://github.com/abey79/vpype/tree/pyside6-migration

The tricky part is the interface with ModernGL, which I use for the rendering. One big change between PySide2 and PySide6 is the transition from `QGLWidget` (now deprecated) to `QOpenGLWidget`, which is trickier to interface to external renderer. I have yet to figure out how to achieve this, and the resources are rather scarce.

**Comment by abey79** [comment]
Migration to PySide6 is now completed.
